src/main/starlark/builtins_bzl/common/cc/cc_helper.bzl

```diff
@@ -1186,6 +1216,12 @@ def _should_use_pic(ctx, cc_toolchain, feature_configuration):
             cc_common.is_enabled(feature_configuration = feature_configuration, feature_name = "prefer_pic_for_opt_binaries")
         )
     )
+def _check_cpp20_module(ctx, feature_configuration):
+    if len(ctx.files.module_interfaces) > 0 and not cc_common.is_enabled(
```

Comment:
Please introduce also a new flag `--experimental_cpp20_modules` into CppOptions, expose it to builtins using a StarlarkMethod in CppConfiguration and verify here that the flag is set.

feature_configuration are technically a flag, and the rule will error out if `--features=cpp20_module` isn't set, however feature flag are relatively hard to track throughout the code, because they are string based without any validation. 

Native flags on the other hand are easlier to track through reference.